images/nginx/rootfs/Dockerfile

```diff
@@ -51,6 +51,7 @@ RUN apk update \
   tzdata \
   grpc-cpp \
   libprotobuf \
+  abseil-cpp-crc-cpu-detect \
```

Comment:
why is this needed? can't recall changes

Reply:
@esigo abseil-cpp-crc-cpu-detect in Dockerfile needed by shared object.

otel_ngx_module.so -> libopentelemetry_exporter_otlp_grpc.so -> libabsl_crc_cpu_detect.so.2308.0.0

Error message clearify this - without libabsl_crc_cpu_detect.so.2308.0.0 you get the errormessage:
```
│ -------------------------------------------------------------------------------                                                                                         │
│   Warning  RELOAD  14s (x16 over 64s)  nginx-ingress-controller  (combined from similar events): Error reloading NGINX:                                                 │
│ -------------------------------------------------------------------------------                                                                                         │
│ Error: exit status 1                                                                                                                                                    │
│ 2024/10/17 13:38:37 [emerg] 49#49: dlopen() "/etc/nginx/modules/otel_ngx_module.so" failed (Error loading shared library libabsl_crc_cpu_detect.so.2308.0.0: No such fi │
│ le or directory (needed by /usr/local/lib/libopentelemetry_exporter_otlp_grpc.so)) in /tmp/nginx/nginx-cfg3367704967:7                                                  │
│ nginx: [emerg] dlopen() "/etc/nginx/modules/otel_ngx_module.so" failed (Error loading shared library libabsl_crc_cpu_detect.so.2308.0.0: No such file or directory (nee │
│ ded by /usr/local/lib/libopentelemetry_exporter_otlp_grpc.so)) in /tmp/nginx/nginx-cfg3367704967:7                                                                      │
│ nginx: configuration file /tmp/nginx/nginx-cfg3367704967 test failed                                                                                                    │
│                                                                                                                                                                         │
│ -------------------------------------------------------------------------------     
``` 
here are the repos, which work "together"
OPENTELEMETRY_CPP_VERSION -> https://github.com/open-telemetry/opentelemetry-cpp/releases
#+export OPENTELEMETRY_CPP_VERSION="v1.17.0"

OPENTELEMETRY_PROTO_VERSION -> https://github.com/open-telemetry/opentelemetry-proto/releases
#+export OPENTELEMETRY_PROTO_VERSION="v1.3.2"

OPENTELEMETRY_CONTRIB_VERSION -> https://github.com/open-telemetry/opentelemetry-cpp-contrib/releases
#+export OPENTELEMETRY_CONTRIB_COMMIT=f6d29426ee9b4d6b476c09ca3cb9bed3cf23906f
OPENTELEMETRY_CONTRIB_COMMIT is the newest commit - an no new version tag is established here. :(

Answer: OPENTELEMETRY_CONTRIB_COMMIT creates libopentelemetry_exporter_otlp_grpc.so, which needs abseil-cpp-crc-cpu-detect.
abseil-cpp-crc-cpu-detect installs the needed libabsl_crc_cpu_detect.so.2308.0.0

Proof:
``` 
src|opentelemetry-cpp-contrib.git $ find . -type f|xargs grep libopentelemetry_exporter_otlp_grpc
./opentelemetry-cpp-contrib.git/instrumentation/otel-webserver-module/build.gradle:    from("${modDepDir}/opentelemetry/${cppSDKVersion}/lib/libopentelemetry_exporter_otlp_grpc.so") { it.into "sdk_lib/lib" }
./opentelemetry-cpp-contrib.git/instrumentation/otel-webserver-module/opentelemetry_module.conf:LoadFile /opt/opentelemetry-webserver-sdk/sdk_lib/lib/libopentelemetry_exporter_otlp_grpc.so
``` 

The other repos do not have a entry like "libopentelemetry_exporter_otlp_grpc".

@strongjz in hope this helps push a faster commit, because:

With azure kubernetes version 1.31 there is a needed pressure using newsest nginx.

Problem: newest nginx uses opentelemetry instead opentracing, but trace status error is not shipping in a correct way.

Due to need use correct monitoring with tracing there's a stop according go to newer versions with nginx.

An this stopps using next kubernetes version 1.31, which makes the pressure from azure.

That's why i created the pull request: https://github.com/kubernetes/ingress-nginx/pull/12371

How can someone push and go to a faster with minimal time delay integrate such a change of 3 lines in build.sh and 1 in Dockerfile in images/nginx/rootfs/ ?
Can you help?
THX a lot all for help!